tensorflow/lite/kernels/internal/reference/space_to_batch_nd.h

```diff
@@ -41,18 +41,10 @@ inline void SpaceToBatchND(const SpaceToBatchParams& params,
                    unextended_output_shape.DimensionsCount());
 
   // Extends the input/output shape from 3D to 4D if needed, NHC -> NH1C.
-  auto extend_shape = [](const RuntimeShape& shape) {
-    if (shape.DimensionsCount() == 4) {
-      return shape;
-    }
-    RuntimeShape new_shape(4, 1);
-    new_shape.SetDim(0, shape.Dims(0));
-    new_shape.SetDim(1, shape.Dims(1));
-    new_shape.SetDim(3, shape.Dims(2));
-    return new_shape;
-  };
-  const RuntimeShape input1_shape = extend_shape(unextended_input1_shape);
-  const RuntimeShape output_shape = extend_shape(unextended_output_shape);
+  const RuntimeShape input1_shape =
+      RuntimeShape::ExtendedShape(4, unextended_input1_shape);
+  const RuntimeShape output_shape =
+      RuntimeShape::ExtendedShape(4, unextended_output_shape);
```

Comment:
is it still worth adding a check for DimensionsCount ==4?

```cc
const RuntimeSHare input1_shape = (unextended_input1_shape.DimensionsCount() == 4) ? unextended_input1_shape : ExtendSHape();
```

Reply:
Done.